core/inputs-common.lua

```diff
@@ -33,7 +33,7 @@ local function debugAST(ast,level)
   if type(ast) == "function" then SU.debug("ast",out.."(function)") end
   if not ast then SU.error("SILE.process called with nil", 1) end
   for i=1, #ast do
-    content = ast[i]
+    local content = ast[i]
```

Comment:
If you are ever rebasing you might throw these scope bugfixes into a different commit in case we cherry pick other pars of the PR without taking the whole thing.

Reply:
Ok, I'll create a smaller PR only with this and HB warning improvements.